gtsfm/retriever/rig_retriever.py

```diff
@@ -20,16 +22,23 @@
 class RigRetriever(RetrieverBase):
     """Retriever for camera rigs inspired by the Hilti challenge."""
 
-    def __init__(self, threshold: int = 100):
+    def __init__(self, subsample: int, threshold: int = 100):
         """Create RigRetriever
 
         Args:
             threshold (int, optional): amount of "proxy" correspondences that will trigger an image-pair. Default 100.
         """
+        self._subsample = subsample
         self._threshold = threshold
 
+    def __accept_constraint(self, constraint: Constraint) -> bool:
+        if not self._subsample:
+            return True
+
+        return constraint.a % self._subsample == 0 and constraint.b % self._subsample == 0
+
     def run(self, loader: LoaderBase) -> List[Tuple[int, int]]:
```

Comment:
should we rename this to run_retreiver? you could also add this to the list of future PRs.

Reply:
Yup naming convention is all over the place. But lets tackle it in a different PR.